boards/stm32f429i-disc1/Kconfig

```diff
@@ -23,7 +23,7 @@ config BOARD_STM32F429I_DISC1
 
     # Put other features for this board (in alphabetical order)
     select HAS_RIOTBOOT
-    select HAS_TINYUSB_DEVICE
+    select HAS_TINYUSB_DEVICE if !BOARD_STM32F429I_DISCO
```

Comment:
Do we still need this with #19007?

Reply:
Hm, unfortunatly it seems so. If you take a look into the [Nightlies output ](https://ci.riot-os.org/details/d11b383ba0cd45048ed6f8d98abbc0de), the compilation of `tests/pkg_tinyusb_cdc_acm_stdio` fails for `stm32f429i-disco` because `usbus*` modules are enabled in Kconfig due to the unconditional settings in the application configuration file [boards/stm32f429i-disco/stm32f429i-disco.config](https://github.com/RIOT-OS/RIOT/blob/master/boards/stm32f429i-disco/stm32f429i-disco.config) https://github.com/RIOT-OS/RIOT/blob/2274a56fef9a8ee780882028d2b361f6675b2a48/boards/stm32f429i-disco/stm32f429i-disco.config#L1-L3
That is, even if we enable `MODULE_STDIO_TINYUSB_CDC_ACM`, all `MODULE_USBUS*` symbols are unconditionally enabled. The only solution for now is to disable the tinyUSB device feature for this board.

This is exactly the same problem I have with all boards that use the single USB interface as `highlevel_stdio` and `usb_board_reset` to control the bootloader mode, for example `boards/common/arduino-mkr`. I can't enable the tinyUSB feature for these boards, although we support `stdio_tinyusb_cdc_acm` and `usb_board_reset` in tinyUSB, because the symbols `MODULE_STDIO_CDC_ACM` and `MODULE_USBUS*` are unconditionally enabled by such application configuration files, for example [boards/common/samdx1-arduino-bootloader/samdx1-arduino-bootloader.config](https://github.com/RIOT-OS/RIOT/blob/master/boards/common/samdx1-arduino-bootloader/samdx1-arduino-bootloader.config) https://github.com/RIOT-OS/RIOT/blob/2274a56fef9a8ee780882028d2b361f6675b2a48/boards/common/samdx1-arduino-bootloader/samdx1-arduino-bootloader.config#L1-L3

I have been trying and thinking about how to solve this problem for days, because this problem is exactly what prevents PR #18998 from being compilable. We could enable the tinyUSB function for all these boards, but the unconditional activation of `MODULE_STDIO_CDC_ACM` and `MODULE_USBUS*` prevents this.

I have already discussed the problem a bit with @MrKevinWeiss. The reason is that the logic of when `MODULE_STDIO_CDC_ACM` and `MODULE_USB*` are enabled in Kconfig is the reverse of the logic in the Makefile dependencies for these modules. In the Makefile dependencies `stdio_cdc_acm` and thus also the `usbus*` modules are activated automatically when needed. It is easy to define an additional prerequisite for them that the `tinyusb_device` is not enabled. In Kconfig, the user has to enable `MODULE_USBUS` first, then he can enable `MODULE_USBUS_CDC_ACM` and then `MODULE_STDIO_CDC_ACM` or even `MODULE_USB_BOARD_RESET` afterwards. The only way to enable them by default if needed is their unconditional activation in such application configuration files which are included in `Makfile.features` for test compilation at the moment.

I personally don't like this logic since the the application/user has to enable them explicitly even if they are required to flash the board at all.

Any suggestion how to solve this problem would be welcome, otherwise we can't enable the tinyUSB feature a large couple of boards.